limitador/src/storage/redis/counters_cache.rs

```diff
@@ -160,10 +176,10 @@ impl Batcher {
         self.notifier.notify_one();
     }
 
-    pub async fn consume<F, Fut, O>(&self, max: usize, consumer: F) -> O
+    pub async fn consume<F, Fut, T, E>(&self, max: usize, consumer: F) -> Result<T, E>
     where
         F: FnOnce(HashMap<Counter, Arc<CachedCounterValue>>) -> Fut,
-        Fut: Future<Output = O>,
+        Fut: Future<Output = Result<T, E>>,
```

Comment:
This signature is getting somewhat … ugly.  But `async` closures aren't a stable thing yet, so I can't have anything call `await` in the closure passed to `consume`… that's how I went about it instead…